orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt

```diff
@@ -60,33 +60,16 @@ interface AuthenticationAware {
   fun retrieveAuthenticatedUser(stage: StageExecution) : PipelineExecution.AuthenticationDetails? {
     return stageNavigator
       .ancestors(stage)
-      .firstOrNull { it.stageBuilder is AuthenticatedStage } ?.let{
-        (it.stageBuilder as AuthenticatedStage).authenticatedUser(solveAuthStages(it.stage)).orElse(null)
+      .firstOrNull {
+        it.stageBuilder is AuthenticatedStage &&
+        it.stage.isManualJudgmentType &&
+        !it.stage.status.isSkipped &&
+        it.stage.withPropagateAuthentication()
+      }?.let{
+        val authStage = it.stage
+        return if(authStage != null) {
+          (it.stageBuilder as AuthenticatedStage).authenticatedUser(authStage).orElse(null)
+        } else null
       }
   }
-
-
-  // When a first valid candidate is found in the ancestors chain is returned
-  // until the ancestor chain was iterated completely at the pipeline beginning
-  fun backtrackStages(stage: StageExecution): StageExecution {
-    if (stage.isManualJudgmentType &&
-      !stage.status.isSkipped &&
-      stage.withPropagateAuthentication()) {
-      return stage;
-    }
-    val previousStage = if (stageNavigator.ancestors(stage).size > 1) stageNavigator.ancestors(stage).get(1).stage else null
-    return if (previousStage == null) stage else backtrackStages(previousStage)
```

Comment:
This line is the core issue - returning stage with the only criteria being that it has no ancestors. This results in any Manual Judgment stage that happens to be the first stage in a pipeline being used to propagate auth, regardless of its state or configuration.